### PR TITLE
Fix python parsing

### PR DESCRIPTION
**Pull request by theHamsta**

### CODE REVIEW

**Comment by theHamsta** [comment]
Rebased #4508 and applied stylua. Pushing to master can always cause a few problems like the closing PR. I think I can reopen the PR when I go back to the commit before it auto-closed. @georgeharker also feel free to force push your original revision https://github.com/nvim-treesitter/nvim-treesitter/commit/4f319e30c63654fa97aaff44a796b15f1447e2bc

**Comment by theHamsta** [comment]
Apart from that formatting diff identical to original PR
```
❯ git diff 4f319e3..HEAD

lua/nvim-treesitter/indent.lua
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

─────────────────────────────────┐
137: function M.get_indent(lnum) │
─────────────────────────────────┘

  while node do
    -- do 'autoindent' if not marked as @indent
    if (not q.indent[node:id()] and not q.aligned_indent[node:id()]
        and q.auto[node:id()] and node:start() < lnum - 1 and lnum - 1 <= node:end_()) then
    if
      not q.indent[node:id()]
      and not q.aligned_indent[node:id()]
      and q.auto[node:id()]
      and node:start() < lnum - 1
      and lnum - 1 <= node:end_()
    then
      return -1
    end


─────────────────────────────────┐
209: function M.get_indent(lnum) │
─────────────────────────────────┘
          local o_srow, o_scol = o_delim_node:start()
          local final_line_indent = false
          if c_delim_node then
              local c_srow, _ = c_delim_node:start()
              if c_srow ~= o_srow and c_srow == lnum - 1 then
                -- delims end on current line, and are not open and closed same line.
                -- final_line_indent controls this behavior, for example this is not desirable
                -- for a tuple.
                final_line_indent = metadata.final_line_indent or false
              end
            local c_srow, _ = c_delim_node:start()
            if c_srow ~= o_srow and c_srow == lnum - 1 then
              -- delims end on current line, and are not open and closed same line.
              -- final_line_indent controls this behavior, for example this is not desirable
              -- for a tuple.
              final_line_indent = metadata.final_line_indent or false
            end
          end
          if final_line_indent then
            -- last line must be indented more in cases where
            -- it would be same indent as next line
            local aligned_indent = o_scol + (metadata.increment or 1)
            if aligned_indent <= indent then
                return indent + indent_size * 1
              return indent + indent_size * 1
            else
                return aligned_indent
              return aligned_indent
            end
          else
            return o_scol + (metadata.increment or 1)
```

File: queries/python/indents.scm [hunk 1]
```diff
@@ -21,24 +16,45 @@
...
+  condition: (parenthesized_expression) @aligned_indent
+  (#set! "delimiter" "()")
+  (#set! "final_line_indent" 1) ; parenthesized_expression already indented
 )
+
 ((ERROR "(" . (_)) @aligned_indent
  (#set! "delimiter" "()"))
 ((argument_list) @aligned_indent
  (#set! "delimiter" "()"))
-((argument_list) @aligned_indent
```
Comment (theHamsta) [inline queries/python/indents.scm:38]: ah, I understand now. Before the query was duplicated :bulb:

**Comment by theHamsta** [comment]
@georgeharker @amaanq anything left to merge this?

File: queries/python/indents.scm (hunk 1 above)
Comment (georgeharker) [reply queries/python/indents.scm:38]: Yep!

**Comment by georgeharker** [comment]
Oh cool, thanks for doing the rebase!

Let me know if there's anything more you'd like me to do.

**Review by georgeharker — APPROVED** [review]

**Pull request merged**